lib/Horde/Crypt/Pgp/Keyserver.php

```diff
@@ -175,7 +175,12 @@ public function getKeyId($address)
                 if (strpos($response->getHeader('Content-Type'), 'text/plain') !== 0) {
                     continue;
                 }
-                $output = $response->getBody();
+                $body = $response->getBody();
+                if (urlencode(urldecode($body)) === $body) {
+                    $output = urldecode($body);
+                } else {
+                    $output = $body;
+                }
```

Comment:
Please note that this should work in theory, but fails sometimes in reality.

I have revert this on my running instance back to

```
-                $output = $response->getBody();
+                $body = $response->getBody();
+                $output = urldecode($body);
```

And I only use a keyserver that returns results urlencoded.